### PR TITLE
#130 Debug mode for touchscreen testing

### LINKED ISSUE #130: Implement 'debugging' mode
https://trello.com/c/BMJhbfeP/425-debugging-mode-for-interactive-labels

When ‘DEBUG=TRUE’ in the env, then allow the following:
- svg 'clickable' regions are shown
- mouse clicker is shown

### PR DESCRIPTION
**Pull request by sam-maher-acmi**
Resolves #130 

Allows for the visualising of touch screen areas/interactions.

### Acceptance Criteria
- [x] Able to see where the user last clicked
- [x] Able to see the clickable regions for the label

### Relevant design files
* None

### Testing instructions
1. Set `DEBUG=True` in your env
2. Load any Interactive Label
3. See that clickable regions are highlighted in green
4. See that a blue dot appears on the last clicked position on the screen
5. Restart the label with `DEBUG=False` and see everything behaves as normal

### CODE REVIEW

File: config.test.env [hunk 1]
```diff
@@ -1,3 +1,4 @@
+DEBUG=False
```
Comment (sighmon) [inline config.test.env:1]: Could you add this to `config.tmpl.env` too please?

File: app/static/debug.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+const root = document.getElementById("root");
```
Comment (sighmon) [inline app/static/debug.js:1]: I'm seeing error: `SyntaxError: Can't create duplicate variable: 'root'`

**Review by sighmon — CHANGES_REQUESTED** [review]
I can see the path outlines, but can't see where I've clicked.

File: app/main.py [hunk 3]
```diff
@@ -73,10 +74,10 @@ def render_playlist():
     for item in list(json_data['playlist_labels']):
         if item['label'] is None:
             json_data['playlist_labels'].remove(item)
-
```
[the remark this answers is not in the document]
Comment (sam-maher-acmi) [reply app/main.py:76]: No 🙃

File: app/static/debug.css [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+}
+
+.screen_touched_icon {
+  pointer-events: none;
+  position: absolute;
+  width: 30px;
+  height: 30px;
+  background-color: rgba(0, 0, 255, 0.5);
+  border-radius: 50%;
+}
```
[the remark this answers is not in the document]
Comment (sam-maher-acmi) [reply app/static/debug.css:13]: 👍

File: config.test.env (hunk 1 above)
Comment (sam-maher-acmi) [reply config.test.env:1]: Of course, good pick up.

File: app/static/debug.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+const debugRoot = document.getElementById("root");
+
+const screenTouchedIcon = document.createElement("div");
+debugRoot.appendChild(screenTouchedIcon);
+screenTouchedIcon.className = "screen_touched_icon";
+
+/** Show where the user last clicked */
+function debugTouchscreenEvent(event) {
+  const mouseX = event.clientX - 15; // Minus half the width of the icon
+  const mouseY = event.clientY - 15; // Minus half the height of the icon
```
Comment (sighmon) [inline app/static/debug.js:10]:
```suggestion
  const mouseX = event.clientX - screenTouchedIcon.offsetWidth / 2;
  const mouseY = event.clientY - screenTouchedIcon.offsetWidth / 2;
```

**Review by sighmon — CHANGES_REQUESTED** [review]
One small change so we don't hardcode values.

**Comment by sam-maher-acmi** [comment]
@sighmon Thanks for the suggestions - I've updated colours to be more accessible.
Referencing [this article](https://davidmathlogic.com/colorblind/#%23648FFF-%23785EF0-%23DC267F-%23FE6100-%23FFB000), I chose colours from the IBM palette. I then adjusted the brightness on the primary blue slightly to pass AAA contrast (against black) - using your [suggested tool](https://webaim.org/resources/contrastchecker/) to verify.

Colours:
Touchable Areas: [#6B95FF](https://convertingcolors.com/hex-color-6B95FF.html)
Clicked Spot: [#FFB000](https://convertingcolors.com/hex-color-FFB000.html)

**Review by sighmon — APPROVED** [review]
Looks great, nice work. Thanks for adding accessibility colours too.

**Pull request merged**